cpp/src/arrow/util/type_fwd.h

```diff
@@ -45,17 +45,18 @@ struct Scope;
 
 struct Compression {
   /// \brief Compression algorithm
+  //   prefix 'ACT'(arrow compression type) to avoid nameing conflicts.
   enum type {
```

Comment:
Can we use `enum class` instead of adding a prefix?

```suggestion
  enum class type {
```

Reply:
OK, let me try.

Reply:
'enum class type {' has the same error if macro definition before, like:`#define SNAPPY 1`.
The error is 'expected identifier before numeric constant'.
Total sample code:
```
#include <iostream>

#define SNAPPY 1

enum class Compression {
  /// \brief Compression algorithm
  UNCOMPRESSED,
  SNAPPY,
  GZIP,
  BROTLI,
  ZSTD,
  LZ4,
  LZ4_FRAME,
  LZO,
  BZ2,
  LZ4_HADOOP
};

int main() {

  std::cout << "UNCOMPRESSED: " << static_cast<int>(Compression::UNCOMPRESSED) << std::endl;
  std::cout << "SNAPPY: " << static_cast<int>(Compression::SNAPPY) << std::endl;

  return 0;
}
```

Reply:
Ref: https://github.com/fmtlib/fmt/commit/8ed264fcd49de5716204e77c6293e1a7aef46906